app/models/validations/setup_validations.rb

```diff
@@ -72,19 +72,15 @@ def validate_irproduct_other(record)
     end
   end
 
-  def validate_scheme_has_confirmed_locations_validation(record)
-    return unless record.scheme
-
-    unless record.scheme.locations.confirmed.any?
-      record.errors.add :scheme_id, :no_completed_locations, message: I18n.t("validations.lettings.setup.scheme.no_completed_locations")
-    end
-  end
-
   def validate_scheme(record)
     if record.scheme&.status == :incomplete
       record.errors.add :scheme_id, :incomplete, message: I18n.t("validations.lettings.setup.scheme.incomplete")
     end
 
+    if record.scheme&.locations.present? && record.scheme.locations.confirmed.none?
+      record.errors.add :scheme_id, :no_completed_locations, message: I18n.t("validations.lettings.setup.scheme.no_completed_locations")
+    end
+
     scheme_during_startdate_validation(record)
     tenancy_startdate_with_scheme_locations(record)
   end
```

Comment:
When a scheme is incomplete, the only error message displayed was “This scheme cannot be chosen as it has no completed locations.” While it is true that a scheme with incomplete details also lacks completed locations (as you cannot begin creating any until the details are complete), this is not the most relevant error message when the user has started but not finished creating a scheme. This change aims to display the “This scheme is incomplete. Select another scheme or update this one.” message and only show the “This scheme cannot be chosen as it has no completed locations.” message when the scheme details are complete but there are no completed locations.